src/pages/search.js

```diff
@@ -40,12 +52,87 @@ export default withSearch(({ search }) => {
     ? ` : ${query} (${totalHits} result${totalHits === 1 ? '' : 's'})`
     : ''
 
-  let searchEvent
+  const getSlug = item => {
+    if (!item.publishDate) {
+      return `${item.slug}`
+    }
+    return `/blog/${item.slug}`
+  }
+
+  const goToResult = selectedItem => {
+    const allResults = [
+      ...results.states.hits,
+      ...results.blogPosts.hits,
+      ...results.pages.hits,
+    ]
+
+    const item = allResults.find(result => {
+      return result.name === selectedItem || result.title === selectedItem
+    })
+
+    if (item && typeof window !== 'undefined') {
+      window.location = getSlug(item)
+    }
+    return false
+  }
 
   return (
     <Layout title="Search">
-      <h2 className="hed-primary">Search{hitsInfo}</h2>
-      <form>
+      <h2 className="hed-primary">Autocomplete Search{hitsInfo}</h2>
+
+      <Combobox
+        openOnFocus
+        onSelect={selectedItem => {
+          goToResult(selectedItem)
+        }}
+      >
+        <ComboboxInput
+          id="ctp-search"
+          placeholder="Search"
+          autoComplete="off"
+          onKeyDown={event => {
+            if (event.key !== 'Enter') {
+              return
+            }
+            if (results && results.length === 1) {
+              goToResult(event.currentTarget.value)
+            }
+          }}
+          onChange={event => {
+            setQuery(event.currentTarget.value)
+          }}
+        />
+        {totalHits ? (
+          <ComboboxPopover id="search-results-popover">
+            {totalHits > 0 ? (
+              <ComboboxList aria-label="Results">
+                {results.states.hits.slice(0, 10).map(state => (
+                  <ComboboxOption
+                    key={`${state.slug}`}
+                    value={`${state.name}`}
+                  />
+                ))}
+                {results.blogPosts.hits.slice(0, 10).map(post => (
+                  <ComboboxOption key={`${post.slug}`} value={post.title} />
+                ))}
+                {results.pages.hits.slice(0, 10).map(page => (
+                  <ComboboxOption key={`${page.slug}`} value={page.title} />
+                ))}
+              </ComboboxList>
+            ) : (
+              <span style={{ display: 'block', marginTop: 5 }}>
+                No results found
+              </span>
+            )}
+          </ComboboxPopover>
+        ) : (
+          false
+        )}
+      </Combobox>
+      <form
+        style={{ marginTop: 50, paddingTop: 50, borderTop: '1px solid black' }}
```

Comment:
@jasonlcrane check my latest commits, I'm using the `.modules.scss` approach like elsewhere in the project and it's really cool

Reply:
Thanks for the reminder ... I was thinking this was temporary just while we play around with it, but my approach here should definitely be avoided.

Reply:
Updated to use the .module.scss file 👍